scipy/interpolate/_cubic.py

```diff
@@ -330,3 +337,204 @@ def from_spline(cls, tck, extrapolate=None):
         raise NotImplementedError("This method does not make sense for "
                                   "an Akima interpolator.")
+
+
+class CubicSpline(PPoly):
+    """Cubic spline data interpolator.
+
+    Interpolate data with a piecewise cubic polynomial which is twice
+    continuously differentiable [1]_. The result is represented as a `PPoly`
+    instance with breakpoints matching the given data.
+
+    Parameters
+    ----------
+    x : array_like, shape (n,)
+        1-d array containing values of the independent variable.
+        Values must be real, finite and in strictly increasing order.
+    y : array_like
+        Array containing values of the dependent variable. It can have
+        arbitrary number of dimensions, but the length along `axis` (see below)
+        must match the length of `x`. Values must be finite.
+    axis : int, optional
+        Axis along which `y` is assumed to be varying. Meaning that for
+        ``x[i]`` the corresponding values are ``np.take(y, i, axis=axis)``.
+        Default is 0.
+    extrapolate : bool, optional
+        Whether to extrapolate to ouf-of-bounds points based on first and last
+        intervals, or to return NaNs. Default is True.
+
+    Attributes
+    ----------
+    x : ndarray, shape (n,)
+        Breakpoints. The same `x` which was passed to the constructor.
+    c : ndarray, shape (4, n-1, ...)
+        Coefficients of the polynomials on each segment. The trailing
+        dimensions match the dimensions of `y`, excluding `axis`. For example,
+        if `y` is 1-d, then ``c[k, i]`` is a coefficient for
+        ``(x-x[i])**(3-k)`` on the segment between ``x[i]`` and ``x[i+1]``.
+    axis : int
+        Interpolation axis. The same `axis` which was passed to the
+        constructor.
+
+    Methods
+    -------
+    __call__
+    derivative
+    antiderivative
+    integrate
+    roots
+
+    See Also
+    --------
+    Akima1DInterpolator
+    PchipInterpolator
+    PPoly
+    InterpolatedUnivariateSpline
+
+    Notes
+    -----
+    Two additional equations are required to determine all coefficients
+    of polynomials on each segment. They are formed from the conditions that
+    at the first and the last interior knots the third derivative is
+    continuous. It essentially means that the first and the second segments
+    (on both ends) are described by the same cubic polynomial. These conditions
+    are called "not-a-knot" and are known to give better interpolation accuracy
+    when nothing is known about end point derivatives [2]_.
+
+    When n=2 or n=3, the solution is sought as a linear/quadratic function
+    passing through the given points.
+
+    `InterpolatedUnivariateSpline` is another class for interpolation based
+    on B-splines. It gives the same interpolation curve, but it selects knots
+    (breakpoints) different from the given `x` and doesn't directly provide
+    coefficients of polynomials on each segment.
```

Comment:
This paragraph can be improved. How about just """InterpolatedUnivariateSpline and `splrep(x, y, s=0)` construct an equivalent interpolation curve in the B-spline basis."""